cogs/onmyoji.py

```diff
@@ -285,6 +323,376 @@ def shiki_bounty_embed(self, shiki):
     async def tengu(self, ctx):
         """hurrhurrhurr"""
         await ctx.send(file=discord.File('./images/tengu.jpg'))
+
+#=====================Shard Trading===========================
+
+    def shard_load_json(self):
+        """
+        Loads the json shard database file. 
+        Generates a new empty one if it does not exist.
+        """
+        try:
+            try:
+                with open(f'{config.list_path}/shard-trading-db.json', 'r') as shard_file:
+                    self.shard_trading_db = json.loads(shard_file.read())
+            except ValueError:
+                self.shard_trading_db = {}
+        except FileNotFoundError:
+            with open(f'{config.list_path}/shard-trading-db.json', 'w') as shard_file:
+                self.shard_trading_db = {}
+                print('New Json file generated!')
+        return
+
+    def shard_file_writeout(self):
+        """Writes to Json shard database file."""
+        with open(f'{config.list_path}/shard-trading-db.json', 'w+') as shard_file:
+            json.dump(self.shard_trading_db, shard_file, indent=4)
+        return
+
+    def shard_split_variable(self, arg):
+        """
+        Splits the incoming '<Shiki> <Amount>' variable into its number and shiki components.
+        """
+        numbers = ''.join(b for b in arg if b.isdigit())
+        shiki = ''.join(b for b in arg if not b.isdigit())
+        return numbers, shiki
+
+    def shard_print_need_list(self, user):
+        """
+        Returns the entirety of the 'need' list of the given user in the format of
+        <Amount> <Shiki>
+        Each on a new line.
+        """
+        need_list = []
+        for i in self.shard_trading_db[user]['need']:
+            if not i:
+                continue
+            numbers, shiki = self.shard_split_variable(i)
+            need_list.append(f"{numbers} {shiki}")
+        self.shard_load_json()
+        return need_list
+        
+
+    def shard_print_have_list(self, user):
```

Comment:
this and the previous functions could be combined
```suggestion
    def shard_print_have_list(self, user, list_name):
    ...
        for i in self.shard_trading_db[user][list_name]
```

Reply:
U rite